src/main/java/io/beanmapper/strategy/MapToRecordStrategy.java

```diff
@@ -0,0 +1,158 @@
+        } catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
+            throw new BeanInstantiationException(targetConstructor.getDeclaringClass(), e);
+        }
+    }
+
+    private <S> Object getValueFromField(final Field field, final S source) {
+        try {
+            if (!field.canAccess(source))
+                field.setAccessible(true);
+            return field.get(source);
+        } catch (IllegalAccessException ex) {
+            throw new RuntimeException("Could not access field " + field.getName());
+        }
+    }
+
+    private <R extends Record> Object getConstructorBasedOnFieldsInSource(final Object source) {
+        Field[] sourceFields = source.getClass().getDeclaredFields();
+        Constructor<R> targetConstructor;
+        try {
+            targetConstructor = getSuitableConstructor(sourceFields, this.getConfiguration().getTargetClass());
+        } catch (NoSuchMethodException e) {
+            throw new BeanInstantiationException(this.getConfiguration().getTargetClass(), e);
+        }
+        return targetConstructor;
+    }
+
+    private <R extends Record> Constructor<R> getSuitableConstructor(Field[] sourceFields, Class<R> targetClass) throws NoSuchMethodException {
+        List<Constructor<R>> constructors = new ArrayList<>(Records.getConstructorsAnnotatedWithRecordConstruct(targetClass));
+        List<Constructor<R>> mandatoryConstructor = constructors.stream()
+                .filter(constructor -> constructor.getAnnotation(RecordConstruct.class).constructMode() == RecordConstructMode.FORCE)
+                .toList();
+        if (!mandatoryConstructor.isEmpty()) {
+            if (mandatoryConstructor.size() > 1) {
+                throw new RecordConstructorConflictException(constructors);
+            }
+            return mandatoryConstructor.get(0);
+        }
+        if (!constructors.isEmpty()) {
+            // Once we get to this point, the constructors-list can only contain constructors with the
+            // RecordConstructMode.ON_DEMAND-option.
+            // Sorts the list in reverse, to make constructor with the most arguments the first to be considered.
+            constructors.sort((arg0, arg1) -> Integer.compare(arg1.getParameterCount(), arg0.getParameterCount()));
+            return getConstructorWithMostMatchingParameters(constructors, sourceFields)
+                    .orElse(Records.getCanonicalConstructorOfRecord(targetClass));
+        }
+        return Records.getCanonicalConstructorOfRecord(targetClass);
+    }
+
+    private <R extends Record> Optional<Constructor<R>> getConstructorWithMostMatchingParameters(List<Constructor<R>> constructors, Field[] sourceFields) {
+        for (var constructor : constructors) {
+            RecordConstruct annotation = constructor.getAnnotation(RecordConstruct.class);
+            List<Field> relevantFields = new ArrayList<>();
+            for (var fieldName : annotation.value()) {
```

Comment:
Persoonlijk vind ik het gebruik van streams in dit geval heel veel leesbaarder. Beetje personal preference though, maar mocht je het iets vinden:
![image](https://user-images.githubusercontent.com/105443742/197715241-48dd6f51-34b0-47bf-ad2e-e5e978b3eba8.png)

Reply:
Pak het nu ook met Streams aan, maar net even iets anders. Heb sourceFields naar een Map verandert, dus is nu met Streams een one-liner. Denk dat dat ook net iets efficiënter is dan de huidige methode.